From 9a440ba5917ff93c6a78726380c6267d91c13f19 Mon Sep 17 00:00:00 2001
From: Alex <alex.shaw.as@gmail.com>
Date: Mon, 30 Sep 2019 16:50:16 -0400
Subject: [PATCH] Prevent duplicate session name.

Several desktop sessions (e.g. KDE Plasma) already include the string " (Wayland)" in the session name. When this happens, the session name displayed to the user is "Plasma (Wayland) (Wayland)". This change makes it so that only "Plasma (Wayland)" will be displayed.

(cherry picked from commit f9a0648ff3f841b68a07c139c3540b9e75bfd279)
---
 src/common/Session.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/Session.cpp b/src/common/Session.cpp
index aa4dad3..923ecd4 100644
--- a/src/common/Session.cpp
+++ b/src/common/Session.cpp
@@ -169,7 +169,7 @@ namespace SDDM {
 
             if (line.startsWith(QLatin1String("Name="))) {
                 if (type == WaylandSession)
-                    m_displayName = QObject::tr("%1 (Wayland)").arg(line.mid(5));
+                    m_displayName = QObject::tr("%1").arg(line.mid(5));
                 else
                     m_displayName = line.mid(5);
             }
-- 
2.38.1

